### ISSUE TITLE
Object.assign on PropTypes

### ISSUE THREAD

**Issue by andreolvr** [yannickcr/eslint-plugin-react#1541]
When i do this: ` return Object.assign({}, propTypes, { dispatch: PropTypes.func.isRequired });` in my propTypes method, i get `react/prop-types` errors, like this: `error 'dispatch' is missing in props validation  react/prop-types`.

Why does it happen? Is this a bug or something?

**Comment by ljharb** [comment]
What happens if you use object spread:
```js
return { ...propTypes, dispatch: PropTypes.func.isRequired };
```
?

In general, it's pretty tough for a static analysis tool to figure things out when you use metaprogramming.

**Comment by andreolvr** [comment]
It works with object spread, but actually, object spread is not an ES feature. Anyway, i'm using it now, thanks!

**Comment by ljharb** [comment]
It's a stage 3 proposal, which will likely land in ES 2018, and is shipping in multiple browsers already, so I'd say it's safe to use :-)
